pkg/oauth/external/handler.go

```diff
@@ -139,6 +139,15 @@ func (h *Handler) AuthenticatePassword(ctx context.Context, username, password s
 	return identitymapper.ResponseFor(h.mapper, identity)
 }
 
+// sanitizeURL removes possible sensitive data from the URL, useful for logging
+func sanitizeURL(u url.URL) *url.URL {
+	u.RawQuery = ""
+	u.RawFragment = ""
+	u.User = nil
```

Comment:
Value receiver functions are only doing shallow copies. I would not expect that the root objects' `*UserInfo` gets nullified in a function called `sanitizeURL`.

Reply:
This is a backport, there shall be no additional changes.

The behavior matches the godoc. Also, the root object's UserInfo is not getting nullified.

Reply:
Oh sorry, I misinterpreted it.